Updates to the warlock profile and lua, updated the priest.lua

Smufrik opened the pull request:
Tweeks to the warlock profiles and Rewrite the priest.lua for cataclysm with changed talents, glyphs and abilitys. Also added a beta shadow profile

Smufrik committed:
Update Warlock.lua

Updated a typo on bane of doom

Smufrik committed:
Update Warlock.lua

Added a first draft of a beta profile for Affliction

Smufrik committed:
Cata Priest release

- Updated talent information
- Included new spells in the game
- Revised ability IDs to reflect changes in Cataclysm
- Added a beta profile specifically for the shadow class

Smufrik committed:
Warlock updates

Intelligent Pet Summoning: The profile dynamically chooses the optimal demon to summon based on the settings.pet_twisting option. This ensures the maximum uptime of the powerful Demon Soul buff through the Felguard or the Felhunter.

Robust Pre-Combat Setup: The pre-combat routine has been significantly enhanced. It intelligently applies Fel Armor and Soul Link only if they are not already active, preventing unnecessary spell casts. It also optimizes the use of the Volcanic Potion by waiting until Metamorphosis is active, maximizing the potion's effectiveness during the high-damage phase.

Dynamic Resource Management: The profile proactively manages resources like mana and Soul Shards. Life Tap is strategically integrated into the rotation when mana is low and Metamorphosis is not coming off cooldown soon. Additionally, Soul Harvest is used judiciously, only when the Warlock has fewer than three Soul Shards, optimizing resource generation.

Cooldown Optimization: The use of Synapse Springs has been introduced to take advantage of longer Metamorphosis cooldowns, ensuring that this powerful cooldown is utilized efficiently.

Demon Soul Enhancement: The profile now leverages Soulburn more effectively. It intelligently consumes a Soul Shard to summon the Felguard when the Demon Soul buff is about to expire, maximizing the duration of this crucial damage increase.

Advanced AoE Handling: The AoE rotation has been improved significantly. It now dynamically switches between Immolation Aura and Rain of Fire based on the number of targets, ensuring optimal damage output in various scenarios. Additionally, Felstorm is used only when a pet is active, preventing errors and ensuring optimal pet utility.

Streamlined Single-Target Rotation: The core single-target rotation has been refined for smoother execution. The Shadow Bolt filler condition has been simplified, prioritizing it when no DoTs are ticking or when Shadow and Flame is about to expire. Bane of Agony has also been integrated to maximize damage on shorter encounters.

Flexible Curse Management: The profile dynamically handles curses, selecting either the single-target or group version of Curse of the Elements based on the curse_grouped setting, adapting to different encounter types.

Smufrik committed:
Update Priest.lua

Updated glyphs

Smufrik committed:
Update WarlockDemonologybeta.simc

Smufrik committed:
Create WarlockAfflictionBeta

Added apl for Affliction

Smufrik committed:
Rename WarlockDemonologybeta.simc to WarlockDemonology.simc

Smufrik committed:
Rename WarlockAfflictionBeta to WarlockAfflictionBeta.simc

Smufrik committed:
Update Warlock.lua

Smufrik committed:
Create PriestShadowBeta.simc

Smufrik committed:
Update Rogue.lua

Added functions from retail assasination.lua and updated variables in the rotations based on theorycrafting forums

Smufrik committed:
Update RogueAssasination.simc

Smufrik committed:
Update RogueAssasination.simc

Smufrik committed:
Update Rogue.lua

Smufrik committed:
Update Rogue.lua

Hekili commented on Cataclysm/Priest.lua:
```diff
@@ -520,42 +541,69 @@ spec:RegisterAuras( {
+        startsCombat = false,
+        texture = 463560,
+
+        handler = function ()
```
This is pseudocode that won't work in the addon.

Hekili commented on Cataclysm/Priest.lua:
```diff
@@ -1541,6 +1603,29 @@ spec:RegisterAbilities( {
+            local totalDamage = baseDamage + bonusDamage
+
+            -- Inflict damage to the target
+            ApplyDamage("target", totalDamage, "shadow")
```
This is pseudocode that won't work in the addon.

Hekili commented on Cataclysm/Priest.lua:
```diff
@@ -1541,6 +1603,29 @@ spec:RegisterAbilities( {
+                -- Target is killed, do nothing
+            else
+                -- Caster takes damage equal to the damage inflicted upon the target
+                ApplyDamage("player", totalDamage, "shadow")
```
This is pseudocode that won't work in the addon.

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -138,7 +141,9 @@ spec:RegisterGlyphs( {
 
 -- Auras
-spec:RegisterAuras( {
+elusiveness = { 247, 2, 13981, 14066 },
```
?

Smufrik replied:
no idea where that came from

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -515,15 +546,232 @@ spec:RegisterAuras( {
+    return 0
+end )
+
+spec:RegisterStateExpr( "effective_combo_points", function ()
```
This always returns `c`.

Smufrik committed:
Update Rogue.lua

updated based on comments

Smufrik committed:
Update Priest.lua

updates based on comments

Smufrik committed:
Merge branch 'Hekili:cataclysm' into cataclysm

Smufrik committed:
-Fixed logic for backstab

Smufrik committed:
-Added Felflame  4pc prock

Smufrik committed:
Update Warlock.lua

fixed q typo

Smufrik committed:
Updated logic for affliction rota

Smufrik committed:
updated curse settings

update to the curse settings for cata since we can use banes and curses now

Smufrik committed:
Hotfix

--Fixed where the rota semed to get stuck on shadowbolt in affliction
--Added logic to demonology aoe rotation

Smufrik committed:
Update Warlock.lua

added curse of guldan

Smufrik committed:
Destruction beta release

Smufrik committed:
Update affliction

updated logic for the profile clipping unstable affliction

Smufrik committed:
Update Warlock.lua

Smufrik committed:
Update Priest.lua

Added power word fortune
changed some ids that were wrong
optimization to the shadow profile

Smufrik committed:
Update PriestShadowBeta.simc

Smufrik committed:
Create WarlockDestructionbeta.simc

Smufrik committed:
Update WarlockDemonology.simc

Smufrik committed:
Update WarlockAfflictionBeta.simc

Smufrik committed:
Update Warlock.lua

Hekili commented on Cataclysm/APLs/PriestShadowBeta.simc:
No need to add Beta to the name if it's going to be the default.

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -1,13 +1,18 @@
 actions+=/variable,name=single_target,value=spell_targets.fan_of_knives<2
 actions+=/variable,name=cold_blood_casted,value=1,if=prev_gcd.1.cold_blood
 actions+=/variable,name=priority_rotation,value=priority_rotation
+actions+=/variable,name=not_pooling|debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|(buff.envenom.up&buff.envenom.remains<=2)|energy.pct>=80|fight_remains<=90,value_else=|debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|energy.pct>=80
```
This variable won't work as written.  It only has a `name` and `value_else`.  Is this supposed to have an `if` (or `condition`) and a default `value`?

Smufrik replied:
missed a if= and (

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -16,32 +21,37 @@ actions+=/call_action_list,name=cds
 actions+=/call_action_list,name=defensives
 
-actions.st+=/variable,name=use_filler,value=combo_points.deficit>1|!variable.single_target
+actions.st+=/variable,name=use_filler,value=combo_points.deficit>1|variable.not_pooling||!variable.single_target
```
Remove extra |.

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -16,32 +21,37 @@ actions+=/call_action_list,name=cds
+actions.st+=/variable,name=use_filler,value=combo_points.deficit>1|variable.not_pooling||!variable.single_target
 actions.st+=/garrote,if=stealthed.all
-actions.st+=/rupture,if=buff.overkill.up&dot.rupture.remains<tick_time|target.time_to_die>6&dot.rupture.remains<tick_time
+actions.st+=/rupture,if=combo_points>1&buff.overkill.up &refreshable&target.time_to_die-remains>20 | combo_points>1&target.time_to_die > 6 &refreshable
```
Remove spaces.

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -16,32 +21,37 @@ actions+=/call_action_list,name=cds
+actions.cds+=/synapse_springs,if=buff.swordguard_embroidery.down&debuff.vendetta.up
+
 actions.cds+=/call_action_list,name=misc_cds
+actions.cds+=/call_action_list,name=items
```
Is there an `items` list in here somewhere that I overlooked?

Smufrik replied:
it was more of a preperation but was forgoten in there

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -16,32 +21,37 @@ actions+=/call_action_list,name=cds
 actions+=/shiv,if=debuff.dispellable_enrage.up
 actions+=/call_action_list,name=defensives
```
This should probably be gated behind a condition that `toggle.defensives` is active to stay aligned with the addon's damage-first design mentality for DPS characters.

Hekili commented on Cataclysm/Priest.lua:
```diff
@@ -1731,7 +1800,7 @@ spec:RegisterOptions( {
 
 -- Packs
--- spec:RegisterPack( "Shadow", 20231124, [[Hekili:vN1wVTTnu4FlbfWWg1vXxAU0bR8W2EynpemaTNLeTeDmr0nirfpdeOF77qsDHsIK2UnDffOOjrMN7NZ35JYUlD)hxNqef7(0QfRwVC5QVyT8(f3S4wxh6XmSRtgk4f0ZWVKGIH)3zpkm9a7XhJsrHmXlslZdGp6q6HcsCH1Ze6(YTwK0k)jv(Wt3JrHwbPXxFiLg9IRZ2sse9RjUBvz6pV6lGkZWbUpDJRZEsyiwCsCrqJ5R8)7CcUGw5NLtsZj0Jv(re2FVlnVY)VWVqIiUoShvW9qsYZry43EIhV4e02iCO7V76eacJZjixNxrWpGhBrjXypAQxibx5VXUYhfqjPjwXKKq ...
+spec:RegisterPack( "ShadowBeta", 20240621, [[Hekili:DJ1AVTTnx4FlbdiTfR1Z2PoTRZoaPdB7nbyzdvPyFyOYIwIkIlsIAKujZab63(ohkzDHIsrPTF5DiOn2sh(CU)WZjUlCV21jGOOUxTC(Yxp)nZF7SLlNVy1sxh1(mQRtgX)wYnWhsjjW)7erc43)EQIGVAFmNeGqi55cF41Uo7YzXQlsD3zf3tCDi5QiUaqkjpuWU11jIfeqlpav676CDetw4H)Ju4vP(cpEi8DFfJNw4fZKk41HCrH3)JEllMnZ1r)q0wiCk8RR0UgnLSlMg4((sWfSmebxNZ5)uH3h4ks5xlb21jHLgSvsjcxfy4gaaNwrfmWXveXnu1Sa6U8WWzsDiz7 ...
```
Just call it Shadow imo.

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -515,15 +544,226 @@ spec:RegisterAuras( {
+
+local function ApplyBleed( key, target )
+    local bleed = tracked_bleeds[ key ]
+    bleed.haste[ target ]        = 100 + GetHaste()
```
Don't use raw API calls since we could theoretically change haste values in a set of recommendations.  Use `state.stat.haste` or whatever is appropriate unless I'm overlooking something.

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -1291,10 +1520,15 @@ spec:RegisterAbilities( {
-        usable = function() return combo_points.current > 0, "requires combo_points" end,
+        usable = function ()
+            if combo_points.current == 0 then return false, "requires combo_points" end
+            if ( settings.rupture_duration or 0 ) > 0 and target.time_to_die < ( settings.rupture_duration or 0 ) then return false, "target will die within " .. ( settings.rupture_duration or 0 ) .. " seconds" end
```
Is this just copied from Retail?  I don't see a `rupture_duration` setting here.

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -515,15 +544,226 @@ spec:RegisterAuras( {
     return combo_points.max
 end )
 
+spec:RegisterSetting( "envenom_pool_pct", 60, {
```
Settings go at the end of the file, before priority imports.

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -515,15 +544,226 @@ spec:RegisterAuras( {
+    return energy.max * ( ( 100 - ( settings.envenom_pool_pct or 100 ) ) / 100 )
+end )
 
+spec:RegisterSetting( "dot_threshold", 7, {
```
Settings go at the end of the file, before priority imports.

Hekili commented on Cataclysm/Rogue.lua:
```diff
@@ -515,15 +544,226 @@ spec:RegisterAuras( {
+    width = "full"
+} )
 
+spec:RegisterSetting( "solo_vanish", true, {
```
Settings go at the end of the file, before priority imports.

Hekili reviewed (changes requested):
See comments for minor changes requested throughout.

Smufrik committed:
Update RogueAssasination.simc

Fixes based on reviews

Smufrik committed:
Rename PriestShadowBeta.simc to PriestShadow.simc

Smufrik committed:
Update Rogue.lua

updated typos in assa profile and moved settings in the code

Smufrik committed:
Update Rogue.lua

Smufrik committed:
Update Priest.lua

Smufrik committed:
Update Rogue.lua

Smufrik committed:
Merge branch 'Hekili:cataclysm' into cataclysm

Smufrik committed:
Rewriten the sub spec

Smufrik committed:
Update Rogue.lua

updated sub rogue with new logic

Smufrik committed:
Update Priest.lua

added dark archangel

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -11,19 +11,19 @@ actions+=/variable,name=regen_saturated,value=energy.regen_combined>settings.reg
 actions+=/variable,name=cold_blood_casted,value=1,if=prev_gcd.1.cold_blood
 actions+=/variable,name=priority_rotation,value=priority_rotation
-actions+=/variable,name=not_pooling|debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|(buff.envenom.up&buff.envenom.remains<=2)|energy.pct>=80|fight_remains<=90,value_else=|debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|energy.pct>=80
+actions+=/variable,name=not_pooling,if=(debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|(buff.envenom.up&buff.envenom.remains<=2)|energy.pct>=80|fight_remains<=90,value_else=|debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|energy.pct>=80
```
We'll still need to revisit this.  There's no `value` being set here, and `value_else` starts with an invalid "or' `|`.  What is this variable intended to be?  Is it meant to always equal `(debuff.shiv.up|cooldown.thistle_tea.full_recharge_time<20)|(buff.envenom.up&buff.envenom.remains<=2)|energy.pct>=80|fight_remains<=90`?  If so, the `if` should be `value` and the `,value_else=...` should be cut.

Hekili commented on Cataclysm/APLs/RogueAssasination.simc:
```diff
@@ -36,14 +36,13 @@ actions.st+=/mutilate,if=variable.use_filler&((target.health.pct>35)|(target.hea
-actions.cds+=/vendetta,if=cooldown.vandetta.charges=2&target.debuff.garrote.duration>6
-actions.cds+=/cold_blood,if=debuff.vendetta.up|target.time_to_die<=120
+actions.cds+=/vendetta,if=cooldown.vendetta.charges=2&target.debuff.garrote.duration>6
+actions.cds+=/cold_blood,if=debuff.vendetta.up&combo_points=5|target.time_to_die<=120& combo_points=5
```
No spaces!

Hekili reviewed (approved):
Approving, but there are still issues to resolve.
